crates/ott-balancer/src/balancer.rs

```diff
@@ -179,6 +179,65 @@ pub struct BalancerContext {
     pub monoliths_by_region: HashMap<String, Vec<MonolithId>>,
 }
 
+pub struct MonolithRegistry {
+    pub monoliths_by_region: HashMap<String, Vec<MonolithId>>,
+    pub monoliths: HashMap<MonolithId, BalancerMonolith>,
+}
+
+pub trait MonolithSelection {
+    
+    fn select_monolith(&self) -> anyhow::Result<&BalancerMonolith>;
+
+    fn random_monolith(&self) -> anyhow::Result<&BalancerMonolith>;
+}
```

Comment:
and then these functions should take something like a `Vec<&BalancerMonolith>` or `impl Iter<&BalancerMonolith>` as input so the `in_region` bits can be deduplicated

Reply:
Just to clarify, `BalancerMonolith` refers to an instance of the Monolith & `BalancerContext` refers to the state the balancer is currently in, with the amount of clients, monoliths, rooms per monolith, etc. Is that right?

Reply:
Yes. 

`BalancerMonolith` directly corresponds to a websocket connection being maintained by the connection manager to the monolith.